src/julia_gc.c

```diff
@@ -647,13 +684,25 @@ static void GapTaskScanner(jl_task_t * task, int root_task)
             rescan = 0;
     }
     if (stack) {
-        if (task->copy_stack) {
+        // task->copy_stack, if not zero or one, is an upper bound for the
+        // size of the stack.
+        if (task->copy_stack > 1) {
```

Comment:
I am not sure this is quite right now -- the value `task->copy_stack == 1` should probably still be treated separately? I think it can only occur for "fresh" tasks which don't have a stack allocation yet; so I assume we could just do `if (task->copy_stack ==1) return;` or so?

Reply:
Hmm, I'll go and have a look at that.

Reply:
So, `task->copy_stack == 1` implies `task->bufsz == 0`. This means that the logic is accidentally correct, but the comment would have to change. But I think that making control flow more explicit by aborting stack scanning for `task->copy_stack == 1` is much clearer, so I'll go with that.